viewing_party/main.py

```diff
@@ -0,0 +1,179 @@
+# --------------------Wave 1--------------------------
+
+def create_movie(movie_title, genre, rating):
+    # create dictionary
+    new_movies = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+    }
+    # if title, genre, rating are true:
+    if movie_title and genre and rating:
+        return new_movies
+    # return dictionary
+    else:
+        return None
+    # else:
+        # return None
+
+def add_to_watched(user_data, movie):
+    # add the new movie to the users data's watched dictionary
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # loop through user_data:
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+        # remove movie_title from watchlist
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    # return user_data
+    return user_data
+
+# # -----------------------Wave 2-------------------------
+
+def get_watched_avg_rating(user_data):
+
+    watched_list = user_data["watched"]
+    rating_list = []
+
+    for movie in watched_list:
+        rating_list.append(movie["rating"])
+    if len(rating_list) == 0:
+        avg_rating = 0.0
+        return avg_rating
+    else:
+        avg_rating = sum(rating_list)/ len(rating_list)
+        return avg_rating
+
+def get_most_watched_genre(user_data):
+
+    most_watched = {}
+    popular_genre = None
+
+    for movie in user_data["watched"]:
+        if movie["genre"] not in most_watched:
+            most_watched[movie["genre"]] = 1
+        else:
+            most_watched[movie["genre"]] += 1
+        popular_genre = max(most_watched, key = most_watched.get)
```

Comment:
Because of the indentation, `popular_genre` is currently being calculated at each iteration of the loop even though you don't need it until the end. 
```suggestion
    popular_genre = max(most_watched, key = most_watched.get)
```